example/cripts/example1.cc

```diff
@@ -0,0 +1,242 @@
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+// The primary include file, this has to always be included
+#include <cripts/Preamble.hpp>
+
+#include <cripts/Bundles/Common.hpp>
+
+// Globals for this Cript
+static Matcher::Range::IP CRIPT_ALLOW({"192.168.201.0/24", "10.0.0.0/8"});
+
+// This is called only when the plugin is initialized
+do_init()
+{
+  TSDebug("Cript", "Hello, example1 plugin is being initialized");
+}
+
+do_create_instance()
+{
+  instance.metrics[0] = Metrics::Sum::create("cript.example1.c0");
+  instance.metrics[1] = Metrics::Sum::create("cript.example1.c1");
+  instance.metrics[2] = Metrics::Sum::create("cript.example1.c2");
+  instance.metrics[3] = Metrics::Sum::create("cript.example1.c3");
+  instance.metrics[4] = Metrics::Sum::create("cript.example1.c4");
+  instance.metrics[5] = Metrics::Sum::create("cript.example1.c5");
+  instance.metrics[6] = Metrics::Sum::create("cript.example1.c6");
+  instance.metrics[7] = Metrics::Sum::create("cript.example1.c7");
+  instance.metrics[8] = Metrics::Sum::create("cript.example1.c8"); // This one should resize() the storage
+
+  Bundle::Common::activate().dscp(10).cache_control("max-age=259200");
+}
+
+do_txn_close()
+{
+  borrow conn = Client::Connection::get();
+
+  conn.pacing = Cript::Pacing::Off;
+  CDebug("Cool, TXN close also works");
+}
+
+do_send_request()
+{
+  borrow req = Server::Request::get();
+
+  req["X-Leif"] = "Meh";
+}
+
+do_read_response()
+{
+  borrow resp = Server::Response::get();
+
+  resp["X-DBJ"] = "Vrooom!";
+}
+
+do_send_response()
+{
+  borrow resp = Client::Response::get();
+  borrow conn = Client::Connection::get();
+  string msg  = "Eliminate TSCPP";
+
+  resp["Server"]         = "";        // Deletes the Server header
+  resp["X-AMC"]          = msg;       // New header
+  resp["Cache-Control"]  = "Private"; // Deletes old CC values, and sets a new one
+  resp["X-UUID"]         = UUID::Unique::get();
+  resp["X-tcpinfo"]      = conn.tcpinfo.log();
+  resp["X-Cache-Status"] = resp.cache;
+  resp["X-Integer"]      = 666;
+  resp["X-Data"]         = AsString(txn_data[2]);
+
+  resp["X-ASN"]         = conn.geo.ASN();
+  resp["X-ASN-Name"]    = conn.geo.ASNName();
+  resp["X-Country"]     = conn.geo.Country();
+  resp["X-ISO-Country"] = conn.geo.CountryCode();
+
+  // Setup some connection parameters
+  conn.congestion = "bbr";
+  conn.dscp       = 8;
+  conn.pacing     = 100000;
+
+  // Some file operations (note that the paths aren't required here, can just be strings, but it's a good practice)
```

Comment:
I assume then that `p3("bar");` would refer to a file named `bar` in the pwd of the traffic server process?

Reply:
Yes, I think so :). It uses std::filesystem